File: backend/src/main/java/com/tjh/riskfactor/service/SectionService.java
package com.tjh.riskfactor.service;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.tjh.riskfactor.entity.form.*;
import com.tjh.riskfactor.repo.SectionRepository;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class SectionService {

    private final SectionRepository repo;

    public Optional<Section> section(Integer id) {
        return repo.findById(id);
    }

    @Transactional
    public List<Section> sectionsOfTask(Integer taskId) {
        var sections = repo.findAllByOwnerTaskId(taskId);
        // getQuestion().size()来触发延迟加载属性的加载动作
        sections.forEach(section ->
            section.getQuestions().size());
        return sections;
    }

}

File: backend/src/main/kotlin/com/tjh/riskfactor/service/SectionService.kt
package com.tjh.riskfactor.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import com.tjh.riskfactor.entity.form.*
import com.tjh.riskfactor.repo.SectionRepository

@Service
class SectionService(override val repo: SectionRepository): IDBService<Section>("section") {

    @Transactional
    fun sectionsOfTask(taskId: Int): List<Section> {
        val sections = repo.findAllByOwnerTaskId(taskId)
        val unused = sections.map { it.questions?.size }.toList()
        return sections
    }

}
